examples/experimental/slpk-in-browser/src/browser-file.ts

```diff
@@ -0,0 +1,132 @@
+// loaders.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+import {FileProvider} from '@loaders.gl/loader-utils';
+
+/**
+ * Provides file data using node fs library
+ * @deprecated - will be replaced with ReadableFile
+ */
+export class BrowserFile implements FileProvider {
+  /** The File object from which data is provided */
+  private file: File;
+
+  /** Create a new BrowserFile */
+  constructor(file: File) {
+    this.file = file;
+  }
+  /**
+   * returns an ArrayBuffer whose contents are a copy of this file bytes from startOffset, inclusive, up to endOffset, exclusive.
+   * @param start The offset, in byte, from the start of the file where to start reading the data.
+   * @param lenght Length of read data
+   */
+  private async getBytesFromFile(start: number, lenght: number): Promise<ArrayBuffer> {
+    let reader = new FileReader();
+    reader.readAsArrayBuffer(this.file.slice(start, start + lenght));
+    return new Promise<ArrayBuffer>((res, rej) => {
+      reader.onload = function() {
+        const arrayBuffer = reader.result
+        if (!arrayBuffer || typeof arrayBuffer === 'string') {
+          rej(new Error('something went wrong'));
+        } else {
+          res(arrayBuffer)
+        }
+      }
+    })
+  }
+
+  /**
+   * Truncates the file descriptor.
+   * @param length desired file lenght
+   */
+  async truncate(length: number): Promise<void> {
+    throw new Error("file loaded in browser cannot be changed");
+  }
+
+  /**
+   * Append data to a file.
+   * @param buffer data to append
+   */
+  async append(buffer: Uint8Array): Promise<void> {
+    throw new Error("file loaded in browser cannot be changed");
+  }
+
+  /** Close file */
+  async destroy(): Promise<void> {
+    throw new Error("file loaded in browser cannot be changed");
+  }
+
+  /**
+   * Gets an unsigned 8-bit integer at the specified byte offset from the start of the file.
+   * @param offset The offset, in bytes, from the start of the file where to read the data.
+   */
+  async getUint8(offset: number | bigint): Promise<number> {
```

Comment:
I feel you should separate this out of your File implementation. 

You read `ArrayBuffer`'s from the FIle, and you create `DataView`s on those ArrayBuffers to read individual binary values.

If you want a help class to wrap DataView that is fine, but I would suggest keeping it separate from `File` to keep things separate and simple